biotrainer/config/configurator.py

```diff
@@ -317,30 +446,127 @@ def _get_config_maps(
+            ConfigurationException:
+                If the `option_name` is not recognized or does not correspond to any known configuration option.
+
+        """
+        if option_name in config_map:
+            config_option = config_map[option_name]
+            config_option.value = value
+            config_option.transform_value_if_necessary(self._config_file_path)
+        else:
+            option_class = all_options_dict.get(option_name)
+            if option_class:
+                config_option = option_class(protocol=self.protocol, value=value)
+                config_option.transform_value_if_necessary(self._config_file_path)
+                config_map[option_name] = config_option
+            else:
+                raise ConfigurationException(f"Unknown configuration option: {option_name}")
+
 
     @staticmethod
-    def _verify_config(
+    def _create_hf_files(
         protocol: Protocol,
         config_map: Dict[str, ConfigOption],
+        hf_map: Dict[str, ConfigOption]
+    ) -> None:
+        """
+        Creates sequences and if needed labels and masks FASTA files based on the HuggingFace
+        dataset configuration and protocol requirements.
+        If the files already exist, they will be overwritten. Subsequently, this method downloads
+        and processes the HuggingFace dataset according to the selected protocol.
+
+        Args:
+            protocol (Protocol): The selected protocol determining how the dataset should be processed.
+            config_map (Dict[str, ConfigOption]): A mapping of configuration option names to their respective ConfigOption instances.
+            hf_map (Dict[str, ConfigOption]): A mapping of HuggingFace dataset option names to their respective ConfigOption instances.
+        Raises:
+            ConfigurationException: If there is an issue during the creation of the required files or processing the dataset.
+                - This can occur if `load_and_split_hf_dataset` fails or if `hf_to_fasta` encounters an error.
+
+        """
+        try:
+            sequences, targets, masks, set_values = load_and_split_hf_dataset(hf_map)
+        except Exception as e:
+            raise ConfigurationException(e)
+
+        if protocol in Protocol.per_sequence_protocols():
+            hf_to_fasta(
+                sequences=sequences,
+                targets=targets,
+                set_values=set_values,
+                sequences_file_name=config_map["sequence_file"].value
+            )
+        elif protocol in Protocol.per_residue_protocols():
+            hf_to_fasta(
+                sequences=sequences,
+                targets=targets,
+                masks=masks,
+                set_values=set_values,
+                sequences_file_name=config_map["sequence_file"].value,
+                labels_file_name=config_map["labels_file"].value,
+                masks_file_name=config_map["mask_file"].value if "mask_file" in config_map else None
+            )
```

Comment:
I think the entire functionality could be done in the utilities file, so you just provide all parameters like sequence_file_name (notice: for consistency, you should use sequence_file_name and mask_file_name) to the `load_and_split_hf_dataset` function, such that we decrease the size of the configurator even more